src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsAfterOneYearValidator.java

```diff
@@ -0,0 +1,411 @@
+                    previousPeriodService.findById(previousPeriodId, request);
+        } catch (MongoException e) {
+
+            throw new DataException(e.getMessage(), e);
+        }
+        if (previousPeriodResponseObject != null && previousPeriodResponseObject.getData() != null) {
+            return previousPeriodResponseObject.getData().getBalanceSheet();
+        } else {
+            return null;
+        }
+    }
+
+    private void crossValidateCurrentPeriodFields(CurrentPeriod currentPeriodCreditors,
+            BalanceSheet currentPeriodBalanceSheet,
+            Errors errors) {
+
+        checkIfCurrentNoteIsNullAndBalanceSheetNot(currentPeriodCreditors, currentPeriodBalanceSheet, errors);
+        checkIfCurrentBalanceSheetIsNullAndNoteNot(currentPeriodCreditors, currentPeriodBalanceSheet, errors);
+        checkIfCurrentBalanceAndNoteValuesAreEqual(currentPeriodCreditors, currentPeriodBalanceSheet, errors);
+    }
+
+    private void checkIfCurrentBalanceAndNoteValuesAreEqual(CurrentPeriod currentPeriodNote,
+            BalanceSheet currentPeriodBalanceSheet,
+            Errors errors) {
+
+        if (!isCurrentPeriodBalanceSheetDataNull(currentPeriodBalanceSheet)
+                && !isCurrentPeriodNoteDataNull(currentPeriodNote)
+                && !(currentPeriodNote.getTotal().equals(currentPeriodBalanceSheet.getOtherLiabilitiesOrAssets()
+                .getCreditorsAfterOneYear()))) {
+
+            addError(errors, currentBalanceSheetNotEqual, CREDITORS_AFTER_CURRENT_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void checkIfCurrentBalanceSheetIsNullAndNoteNot(CurrentPeriod currentPeriodNote,
+            BalanceSheet currentPeriodBalanceSheet,
+            Errors errors) {
+
+        if (isCurrentPeriodBalanceSheetDataNull(currentPeriodBalanceSheet) &&
+                !isCurrentPeriodNoteDataNull(currentPeriodNote)) {
+
+            addError(errors, currentBalanceSheetNotEqual, CREDITORS_AFTER_CURRENT_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void checkIfCurrentNoteIsNullAndBalanceSheetNot(CurrentPeriod currentPeriodNote,
+            BalanceSheet currentPeriodBalanceSheet,
+            Errors errors) {
+
+        if (isCurrentPeriodNoteDataNull(currentPeriodNote) &&
+                !isCurrentPeriodBalanceSheetDataNull(currentPeriodBalanceSheet)) {
+
+            addError(errors, currentBalanceSheetNotEqual, CREDITORS_AFTER_CURRENT_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private BalanceSheet getCurrentPeriodBalanceSheet(HttpServletRequest request,
+            String companyAccountsId) throws DataException {
+
+        String currentPeriodId = currentPeriodService.generateID(companyAccountsId);
+
+        ResponseObject<uk.gov.companieshouse.api.accounts.model.rest.CurrentPeriod> currentPeriodResponseObject;
+
+        try {
+
+            currentPeriodResponseObject = currentPeriodService.findById(currentPeriodId, request);
+        } catch (MongoException e) {
```

Comment:
You shouldn't need to catch a `MongoException` here, currentPeriodService.findById should throw a `DataException`